packages/clerk-js/src/ui/elements/Web3SolanaWalletButtons.tsx

```diff
@@ -0,0 +1,277 @@
+import { WalletReadyState } from '@solana/wallet-adapter-base';
+import { ConnectionProvider, useWallet, WalletProvider } from '@solana/wallet-adapter-react';
+import { MAINNET_ENDPOINT } from '@solana/wallet-standard';
+import type { Ref } from 'react';
+import React, { forwardRef, isValidElement, useMemo } from 'react';
+
+import { ProviderInitialIcon } from '@/ui/common';
+import {
+  Button,
+  descriptors,
+  Flex,
+  Grid,
+  Icon,
+  Image,
+  localizationKeys,
+  SimpleButton,
+  Spinner,
+  Text,
+  useLocalizations,
+} from '@/ui/customizables';
+import { Card } from '@/ui/elements/Card';
+import { useCardState } from '@/ui/elements/contexts';
+import { LinkRenderer } from '@/ui/elements/LinkRenderer';
+import { distributeStrategiesIntoRows } from '@/ui/elements/utils';
+import { mqu, type PropsOfComponent } from '@/ui/styledSystem';
+import { sleep } from '@/ui/utils/sleep';
+
+type Web3WalletButtonsProps = {
+  web3AuthCallback: ({ walletName }: { walletName: string }) => Promise<unknown>;
+};
+
+const SOCIAL_BUTTON_BLOCK_THRESHOLD = 2;
+const SOCIAL_BUTTON_PRE_TEXT_THRESHOLD = 1;
+const MAX_STRATEGIES_PER_ROW = 5;
+
+const Web3SolanaWalletButtonsInner = ({ web3AuthCallback }: Web3WalletButtonsProps) => {
+  const card = useCardState();
+  const { wallets } = useWallet();
+  const { t } = useLocalizations();
+
+  // Filter to only show installed wallets
+  const installedWallets = React.useMemo(
+    () =>
+      wallets
+        .filter(w => {
+          return w.readyState === WalletReadyState.Installed;
+        })
+        .map(wallet => {
+          return {
+            name: wallet.adapter.name,
+            icon: wallet.adapter.icon,
+          };
+        }),
+    [wallets],
+  );
+
+  const startWeb3AuthFlow = (walletName: string) => async () => {
+    card.setLoading(walletName);
+    try {
+      await web3AuthCallback({ walletName });
+    } catch {
+      await sleep(1000);
+    } finally {
+      card.setIdle();
+    }
+  };
+
+  const { strategyRows } = distributeStrategiesIntoRows(installedWallets, MAX_STRATEGIES_PER_ROW, undefined);
+  const strategyRowOneLength = strategyRows.at(0)?.length ?? 0;
+  const shouldForceSingleColumnOnMobile = installedWallets.length === 2;
+  const ButtonElement = installedWallets.length <= SOCIAL_BUTTON_BLOCK_THRESHOLD ? WalletButtonBlock : WalletButtonIcon;
+
+  if (installedWallets.length === 0) {
+    return (
+      <Card.Alert>
+        <LinkRenderer
+          text={t(
+            localizationKeys('web3WalletButtons.noneAvailable', {
+              solanaWalletsLink: 'https://solana.com/solana-wallets',
+            }),
+          )}
+          isExternal
+          sx={t => ({
+            textDecoration: 'underline',
+            textUnderlineOffset: t.space.$1,
+            color: 'inherit',
+          })}
+        />
+      </Card.Alert>
+    );
+  }
+
+  return (
+    <Flex
+      direction='col'
+      gap={2}
+    >
```

Comment:
We are missing element descriptors on all of the elements in here. We likely want to match what is done in SocialButtons?

Reply:
https://github.com/clerk/javascript/pull/7417
Created this separate PR to isolate this fix to merge into this branch